### PR TITLE
Add basic docs on doc provider configuration

### PR DESCRIPTION
**Pull request by starbelly**
- add basic documentation for configuring a doc provider
 - update publishing examples to match current behavior

### CODE REVIEW

**Comment by starbelly** [comment]
So, we're going to add `--without-docs` options, etc. but I wanted to go ahead and get this updated now per releasing 6.11.0

**Comment by starbelly** [comment]
> So, we're going to add `--without-docs` options, etc. but I wanted to go ahead and get this updated now per releasing 6.11.0

Ignore this, I went ahead and opened up a PR for `--without-docs` on the publish task and a `--dry-run` option for the docs task.

**Review by ferd — APPROVED** [review]
Sounds good, though I believe the --dry-run thing is still on a pending PR?

**Comment by starbelly** [comment]
@ferd yeah, I want to get those two merged first, and possibly one more, before merging this.

File: content/en/docs/package_management/hex_package_management.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ See [Base Config/edoc](/docs/configuration/configuration/#edoc) for more informa
 
 ### Without a doc provider
 
- One may choose not to use use edoc to generate their documentation and we support this. In this case none of the above configuration is needed. All one has to do is have a `doc/` directory present in the root of the project directory with an `index.html` file present. 
+ One may choose not to use use edoc to generate their documentation and we support this. In this case, none of the above configuration is needed. All one has to do is have a `doc/` directory present in the root of the project directory with a least an `index.html` file present.
```
Comment (ferd) [inline content/en/docs/package_management/hex_package_management.md:97]:
```suggestion
 One may choose not to use use edoc to generate their documentation and we support this. In this case, none of the above configuration is needed. All one has to do is have a `doc/` directory present in the root of the project directory with at least an `index.html` file present. 
```

File: content/en/docs/package_management/hex_package_management.md [hunk 2]
```diff
@@ -35,11 +35,67 @@ You'll be asked what type of increment to make to the version or it can be suppl
...
+ - edoc 
+
+### Global config
+
+Note that when we mean globally within the scope of a project, since a project may have many reposconfigured. Below is an example of such a configuration: 
+
+```erlang
+{erl_opts, [debug_info]}.
+{deps, []}.
+{hex, {doc, #{provider => edoc}}}.
```
Comment (starbelly) [inline content/en/docs/package_management/hex_package_management.md:69]:
```suggestion
{hex, [{doc, #{provider => edoc}]}}.
```

**Pull request merged**